src/pages/benchviz/Benchmark.tsx

```diff
@@ -0,0 +1,151 @@
+import * as d3 from "d3";
+import React from "react";
+import * as zlib from "zlib";
+
+import { barComparisonGraph } from "./bar-comparison-graph";
+import { positiveNegativeBarGraph } from "./positive-negative-bar-graph";
+import { joinOnProperty, JoinResult } from "./util";
+
+export enum BenchmarkKind {
+    Memory = "memory",
+}
+
+export type BenchmarkResult = MemoryBenchmarkResult;
+
+export enum MemoryBenchmarkCategory {
+    TotalMemory = "totalMemory",
+    Garbage = "garbage",
+}
+
+export interface MemoryBenchmarkResult {
+    kind: string;
+    categories: Record<MemoryBenchmarkCategory, number>;
+    benchmarkName: string;
+}
+
+// Utility functions
+const formatBenchmarkName = (benchmarkName: string) => benchmarkName.replace(".lua", "").split("/").pop()!;
+const formatMemory = (value: number) => `${Math.round(value / 10) / 100} Kb`;
+
+const benchmarkGarbage = (bm: BenchmarkResult) => bm.categories[MemoryBenchmarkCategory.Garbage];
+const garbagePercentChange = (result: JoinResult<BenchmarkResult>) =>
+    (benchmarkGarbage(result.right!) - benchmarkGarbage(result.left!)) / benchmarkGarbage(result.left!);
+
+// Get benchmark data from URL parameter
+const data = window.location.search.split("?d=")[1];
```

Comment:
There is no guarantee that the script would be loaded when you're actually opening the page. No idea how it's done in Docusaurus, but some react frameworks preload scripts when you hover over a link. So generally it's better to do any logic inside the component itself, which is guaranteed to be mounted at the right time.